### ISSUE TITLE
Optimize thread redirect database structure

### ISSUE THREAD

**Issue by PaulBender** [mybb/mybb#2198]
Currently, threads that are left behind as a redirect store their redirect information in the closed column on the threads table in the style as **moved|$tid**. This can slow a site down as a full table scan is needed for queries that need to avoid moved threads.

The following changes should be made:
- Add a moved column to the thread table that stores a moved thread's tid.
- Change the closed column to a tinyint and use 2 for moved threads.

**Comment by Devilshakerz** [comment]
>
- Add a moved column to the thread table that stores a moved thread's tid.
- Change the closed column to a tinyint and use 2 for moved threads.

That would increase the already existing confusion. Wouldn't it be better to compare the new `moved` column against `NULL` (default) instead?

**Comment by Stefan-ST** [comment]
I'm against changing this in a minor release because it requires a lot of changes and may break plugins.

**Comment by JN-Jones** [comment]
I'm against it too as it would mean that we need to update all existing redirects which may be quite a lot. Also plugins may require it as @Stefan-ST said.

@dragonexpert wanted to do the change for one of his plugins, don't know what his results were but probably he can show a few results.

**Comment by Destroy666x** [comment]
Exactly, it will all break plugins which query thread information correctly now, so I wouldn't do it either. It's obviously bad database design (same applies to several comma separated columns) but we should have fixed it in 1.8.0, if anything. And we shouldn't repeat these mistakes in 2.0.

Closing since 2 other people disagreed as well.

**Comment by dragonexpert** [comment]
What I do in my plugin is create a new column called moved and have it be the tid of the redirect.  0 means no redirect.  I then query for the moved column instead of closed.

**Comment by PaulBender** [comment]
Reopening this issue for fixing in 1.10

**Comment by Azareal** [comment]
There probably needs to be some sort of moved column, ideally *without* re-purposing an existing column which has nothing to do with moved topics like closed. If you're worried about space, a TID of 0 in the moved column could signify that it's not moved (as no topic should ever be 0).

I'm slightly worried that null might wind up getting casted to 0 anyway along the line, but maybe it's a non-issue.

**Comment by arcalys** [comment]
@Ben-MyBB 
This can be closed
